### PR TITLE
Add bitcoind v24

### PR DESCRIPTION
**Pull request by benthecarman**
https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.md

### CODE REVIEW

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala [hunk 1]
```diff
@@ -368,10 +370,10 @@ sealed trait BitcoindVersion
 object BitcoindVersion extends StringFactory[BitcoindVersion] with Logging {
 
   /** The newest version of `bitcoind` we support */
-  val newest: BitcoindVersion = V23
+  val newest: BitcoindVersion = V24
```
Comment (Christewart) [inline bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala]: I'm seeing some test failures in `nodeTest`. I'm _guessing_ that its related to bumping this version. 

If we wanted to merge this PR, i would suggest reverting this to `V23` and I can investigate why our `nodeTest` does not seem to be playing nicely with `V24`. 

I believe that will unblock you so you can use v24 while I investigate the bug in our `node` impl. If we take this path, we will need to make sure we file an issue to bump this version and fix the bug in `node`. 

Thoughts?

File: testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala [hunk 2]
```diff
@@ -110,6 +111,7 @@ trait BitcoindRpcTestUtil extends Logging {
                   |port=${uri.getPort}
                   |debug=$isDebug
                   |walletbroadcast=1
+                  |mempoolfullrbf=1
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:114]: 👀 👀 😂

File: bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala (hunk 1 above)
Comment (Christewart) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala]: bug is most likely related to this: https://github.com/bitcoin/bitcoin/blob/master/doc/release-notes/release-notes-24.0.md#p2p-and-network-changes

Comment (benthecarman) [reply bitcoind-rpc/src/main/scala/org/bitcoins/rpc/client/common/BitcoindRpcClient.scala]: I tried debugging for a little but can't figure it out. Went ahead and reverted to v23, if you wanna take a stab at it that'd be great

**Review by Christewart — APPROVED** [review]

**Pull request merged**